app/models/course/enrol_request.rb

```diff
@@ -0,0 +1,25 @@
+class Course::EnrolRequest < ActiveRecord::Base
+  stampable
+  has_paper_trail on: [:destroy]
+  validates :course_id, uniqueness: { scope: :user_id }
```

Comment:
i think this is the other way around. but if we already have a unique constraint on course_id -> user_id, then this is not needed because the validations gem will propagate it for us automatically.